core-common/src/main/java/org/uniprot/core/util/EnumDisplay.java

```diff
@@ -32,29 +41,45 @@
     }
 
     /**
-     * ignore case and trim on passed string argument to match
+     * Returns the Enum constant by its name. String value of name is trimmed and case is ignored
+     * while finding the corresponding enum constant.
      *
-     * @throws IllegalArgumentException for null or not matching argument
+     * @param name string value of the enum to be returned.
+     * @param enumClass the Class object of the enum type
+     * @return Enum constant for the given name and enum type
+     * @throws IllegalArgumentException if either name or enumClass is null. or name doesn't have
+     *     corresponding enum
      */
-    static <T extends Enum<T> & EnumDisplay> T typeOf(@Nonnull String name, Class<T> enumClass) {
+    static <T extends Enum<T> & EnumDisplay> T typeOf(
+            @Nonnull String name, @Nonnull Class<T> enumClass) {
         nullThrowIllegalArgument(name);
+        nullThrowIllegalArgument(enumClass);
 
-        for (T constant : enumClass.getEnumConstants()) {
-            if (constant.getCompareOn().equalsIgnoreCase(name.trim())) return constant;
-        }
-        throw new IllegalArgumentException(
-                "The " + enumClass.getSimpleName() + " with " + name + " doesn't exist");
+        RuntimeException exception =
+                new IllegalArgumentException(
+                        "The " + enumClass.getSimpleName() + " with " + name + " doesn't exist");
+
+        return Arrays.stream(enumClass.getEnumConstants())
+                .filter(constant -> constant.getCompareOn().equalsIgnoreCase(name.trim()))
+                .findAny()
```

Comment:
From java doc
Returns an Optional<T> describing some element of the stream, or an empty Optional<T> if the stream is empty. The behavior of this operation is explicitly nondeterministic; it is free to select any element in the stream. This is to allow for maximal performance in parallel operations;

might be findFirst is better approach?

Reply:
Changed to findFirst